### PR TITLE
chore: update crypto-js to @4 in aws-amplify@latest

### PR DESCRIPTION
**Pull request by iartemiev**
#### Description of changes

* Upgrading CryptoJS to 4
* Adding workaround implementation of `WordArray.random` for RN since it doesn't support latest CryptoJS

#### Issue #, if available

https://github.com/aws-amplify/amplify-js/issues/8256

#### Description of how you validated changes
* Built the lib code, deployed to Verdaccio
* Installed from Verdaccio into RN sample apps that use Auth and DataStore
* Validated that expected functionality is intact in those
* Installed into web sample apps for the same categories and validated the same ^
* Ran e2e tests in CCI. Link to successful run: https://app.circleci.com/pipelines/github/aws-amplify/amplify-js/7877/workflows/42240f50-28a6-467f-b68a-b48ee8a393c7

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/datastore/src/util.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ import {
 	SortPredicatesGroup,
 	SortDirection,
 } from './types';
+import { WordArray } from 'amazon-cognito-identity-js';
```
Comment (manueliglesias) [inline packages/datastore/src/util.ts:22]: should this appear now as a dependency in the package.json? (managed by lerna)

**Review by manueliglesias — APPROVED** [review]
LGTM

File: packages/amazon-cognito-identity-js/package.json [hunk 2]
```diff
@@ -68,7 +68,7 @@
   "types": "./index.d.ts",
   "dependencies": {
     "buffer": "4.9.2",
-    "crypto-js": "^3.3.0",
+    "crypto-js": "^4.0.0",
```
Comment (sammartinez) [inline packages/amazon-cognito-identity-js/package.json:71]: In the version 3 PR, is does not have a `^` do we want to remove the `^` here?

Comment (iartemiev) [reply packages/amazon-cognito-identity-js/package.json:71]: No, that was a fat finger in 3. I'm updating it

File: packages/datastore/src/util.ts (hunk 1 above)
Comment (iartemiev) [reply packages/datastore/src/util.ts:21]: Updated

Comment (sammartinez) [reply packages/datastore/src/util.ts:21]: yeah I was curious on this as well?

**Review by manueliglesias — APPROVED** [review]
LGTM 👍

**Review by amhinson — APPROVED** [review]
LGTM 👍

**Review by sammartinez — APPROVED** [review]
LGTM 🌮

**Pull request merged**